exchange/exchange_test.go

```diff
@@ -1585,6 +1587,53 @@ func TestExchangeJSON(t *testing.T) {
 	}
 }
 
+func TestValidBidID(t *testing.T) {
+	fileName := "./test/bidid/valid-bid-id.json"
+	bid, spec := testBidIdHelper(t, fileName, false)
+	diffJson(t, "Bid Ext info modified", bid.SeatBid[0].Bid[0].Ext, spec.Response.Bids.SeatBid[0].Bid[0].Ext)
+}
+
+func TestInvalidBidID(t *testing.T) {
+	fileName := "./test/bidid/invalid-bid-id.json"
+	bid, spec := testBidIdHelper(t, fileName, true)
+	diffJson(t, "bid.SeatBid[0].Bid[0].Ext modified", bid.SeatBid[0].Bid[0].Ext, spec.Response.Bids.SeatBid[0].Bid[0].Ext)
+	diffJson(t, "bid.Ext info modified", bid.Ext, spec.Response.Ext)
+}
+
+func testBidIdHelper(t *testing.T, fileName string, bidIDError bool) (*openrtb2.BidResponse, *exchangeSpec) {
```

Comment:
Why not use the existing JSON test framework for these tests as well? You can just enable/disable generating Bid ID by setting the `generatBidID` flag in the JSON test it self. See this debug log disabled test for example: https://github.com/prebid/prebid-server/blob/master/exchange/exchangetest/debuglog_disabled.json

Reply:
This a genius thing I didn't know about! Modified!